tests/scene/test_graph_node.h

```diff
@@ -0,0 +1,60 @@
+/**************************************************************************/
+/*  test_graph_node.h                                                        */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_GRAPH_NODE_H
+#define TEST_GRAPH_NODE_H
+
+#include "scene/gui/graph_node.h"
+#include "scene/main/window.h"
+
+#include "tests/test_macros.h"
+
+namespace TestGraphNode {
+
+TEST_CASE("[SceneTree][GraphNode]") {
+	SUBCASE("[GraphNode][Only Child] Graph Node only child on delete should not cause error.") {
+        //Setup
+		GraphNode *test_node = memnew(GraphNode);
+        test_child->set_name("Graph Node");
+		Control *test_child = memnew(Control);
+		test_child->set_name("child");
+		test_node->add_child(test_child);
+
+        //Test
+        CHECK_NOTHROW_MESSAGE(test_node->remove_child(test_child));
+
+		memdelete(test_child);
```

Comment:
You should only need to delete `test_node`, which should delete its child nodes too.